### PR TITLE
Add _domain.isEmpty() procedure

### PR DESCRIPTION
**Pull request by nimitbhardwaj**
With reference to  #7721, I made the pull request, its my first PR on Chapel, if there is any changes (I suppose tests, but I found the necessary test at  test/domains/claridge/DomainIsEmpty.chpl) then please tell me.

### CODE REVIEW

File: modules/internal/ChapelArray.chpl [hunk 1]
```diff
@@ -1650,6 +1650,11 @@ module ChapelArray {
        return translate(offTup);
      }
 
+     /* Return true if the array has no elements */
```
Comment (benharsh) [inline modules/internal/ChapelArray.chpl]: This should say 'domain' instead of 'array'.

**Comment by benharsh** [comment]
I'll start some testing:
- [ ] full local

File: modules/internal/ChapelArray.chpl (hunk 1 above)
Comment (nimitbhardwaj) [reply modules/internal/ChapelArray.chpl]: oh yes, I just copied this comment from array section, I edit it

**Comment by benharsh** [comment]
Sorry, one last request: I think "elements" should be changed to "indices".

**Comment by ben-albrecht** [comment]
@nimitbhardwaj - could you add a simple test to `test/domains/userAPI` for this feature? You could follow `test/arrays/userAPI/arrayAPItest.chpl` as an example.

**Comment by nimitbhardwaj** [comment]
done

File: test/domains/userAPI/isEmptyTestDomain.good [hunk 2]
```diff
@@ -0,0 +1,5 @@
+false
+true
+false
+true
+
```
Comment (benharsh) [inline test/domains/userAPI/isEmptyTestDomain.good]: This extra line will cause the test to fail. You can check this on your end with:
```
start_test isEmptyTestDomain.chpl
```

Comment (nimitbhardwaj) [reply test/domains/userAPI/isEmptyTestDomain.good]: ok

**Pull request merged**